test/src/test/java/org/corfudb/integration/CorfuReplicationClusterConfigIT.java

```diff
@@ -1676,4 +1689,30 @@ private void updateTopology(CorfuStore corfuStore, ClusterUuidMsg op) {
             txn.commit();
         }
     }
+
+    // StreamListener on the ReplicationStatus table which updates the latch when ReplicationStatus reaches STOPPED
+    private class ReplicationStopListener implements StreamListener {
+
+        private CountDownLatch countDownLatch;
+
+        ReplicationStopListener(CountDownLatch countDownLatch) {
+            this.countDownLatch = countDownLatch;
+        }
+
+        @Override
+        public void onNext(CorfuStreamEntries results) {
+            results.getEntries().forEach((schema, entries) -> entries.forEach(e -> {
+                LogReplicationMetadata.ReplicationStatusVal statusVal =
+                    (LogReplicationMetadata.ReplicationStatusVal)e.getPayload();
+                if (statusVal.getStatus().equals(LogReplicationMetadata.SyncStatus.STOPPED)) {
+                    countDownLatch.countDown();
+                }
+            }));
+        }
+
+        @Override
+        public void onError(Throwable throwable) {
```

Comment:
Not sure about these but,
 1. Can we add `countDownLatch.countDown();` - to indicate the `countDownLatch.await()` thread to proceed and fail? Else, it may keep waiting there. and,
2. Can we log an error message/stack trace?

Reply:
1. We should not count it down but throw an exception here.  Valid points about failing and logging the error.